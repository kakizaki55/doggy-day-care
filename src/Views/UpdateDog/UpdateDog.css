.update-dog {
  background-image: url('../../assests/cool-background.svg');
  height: 560px;
  background-color: rgb(172, 66, 172);
}
