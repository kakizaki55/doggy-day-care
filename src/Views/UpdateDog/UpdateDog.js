import React from 'react';
import { useEffect, useState } from 'react';
import { fetchDogById } from '../../services/fetchDogs';
import UpdateDogFrom from '../../components/UpdateDogFrom/UpdateDogForm';
import { deleteDogById, updateDogInDataBase } from '../../services/updateDog';

export default function UpdateDog(props) {
  const [dog, setDog] = useState({});
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  const updateDog = (key, value) => {
    dog[key] = value;
    setDog({ ...dog });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    await updateDogInDataBase(dog);
  };
  const handleDelete = async (e) => {
    e.preventDefault();
    await deleteDogById(dog);
  };

  return (
    <>
      <UpdateDogFrom
        dog={dog}
        updateDog={updateDog}
        handleSubmit={handleSubmit}
        handleDelete={handleDelete}
      />
    </>
  );
}
