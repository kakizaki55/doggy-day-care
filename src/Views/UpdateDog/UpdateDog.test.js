import UpdateDog from './UpdateDog';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';

test('makling sure that my update dog is working correctly', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dog/8']}>
      <Route exact path="/dog/:id" component={UpdateDog}></Route>
    </MemoryRouter>
  );
  await screen.findByDisplayValue('Amely');
  expect(container).toMatchSnapshot();
});
