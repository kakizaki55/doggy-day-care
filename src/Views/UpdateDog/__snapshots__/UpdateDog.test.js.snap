// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makling sure that my update dog is working correctly 1`] = `
<div>
  <div
    class="update-dog"
  >
    <span />
    <h1>
      Update Dog
    </h1>
    <form
      class="update-form-container"
    >
      <div
        class="update-form"
      />
      <label>
        name:
        <input
          placeholder="name"
          value="Amely"
        />
      </label>
      <label>
        age:
        <input
          placeholder="age"
          value="0"
        />
      </label>
      <label>
        Breed:
        <input
          placeholder="breed"
          value="Rottweiler"
        />
      </label>
      <label>
        bio:
        <textarea
          placeholder="bio"
        >
          Highly energetic - loves long walks and tug of war. Expect he'll grow to be a lot bigger than a teacup.
        </textarea>
      </label>
      <label>
        Image URL:
        <input
          placeholder="image url"
          value="https://placedog.net/500?id=12"
        />
      </label>
      <button>
        Update Dog
      </button>
      <button>
        Delete
      </button>
    </form>
  </div>
</div>
`;
